cpu/samd21/periph/i2c.c

```diff
@@ -435,7 +435,7 @@ static inline int _write(SercomI2cm *dev, char *data, int length)
     DEBUG("Looping through bytes\n");
     while (tmp_data_length--) {
         /* Check that bus ownership is not lost. */
-        if (!(dev->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE(2))) {
+        if ((dev->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE_Msk) != SERCOM_I2CM_STATUS_BUSSTATE(2)) {
```

Comment:
Owner is 0b10. If we only check the MSB, we could miss if register was Busy (0b11) as the MSB will still be set.